lib/_tls_wrap.js

```diff
@@ -1476,8 +1476,10 @@ Server.prototype.addContext = function(servername, context) {
     RegExpPrototypeSymbolReplace(/([.^$+?\-\\[\]{}])/g, servername, '\\$1'),
     '*', '[^.]*',
   ) + '$');
-  ArrayPrototypePush(this._contexts,
-                     [re, tls.createSecureContext(context).context]);
+
+  const secureContext =
+    context instanceof common.SecureContext ? context : tls.createSecureContext(context);
```

Comment:
Might be worth implementing a more efficient `isSecureContext(...)` method under `util/types`

Reply:
I don't think this function is in hot path, so might not be worth the perf improvement vs code growth in `primordials`. Not too sure.